File: src/main/java-graphql/io/github/graphql/resolver/CommitResolver.java
package io.github.graphql.resolver;

import io.github.graphql.model.*;

/**
 * Resolver for Commit
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:48+0800"
)
public interface CommitResolver {

    PullRequestConnectionTO associatedPullRequests(CommitTO commitTO, String after, String before, Integer first, Integer last, PullRequestOrderTO orderBy) throws Exception;

    @javax.validation.constraints.NotNull
    BlameTO blame(CommitTO commitTO, String path) throws Exception;

    @javax.validation.constraints.NotNull
    CommitCommentConnectionTO comments(CommitTO commitTO, String after, String before, Integer first, Integer last) throws Exception;

    DeploymentConnectionTO deployments(CommitTO commitTO, String after, String before, java.util.List<String> environments, Integer first, Integer last, DeploymentOrderTO orderBy) throws Exception;

    @javax.validation.constraints.NotNull
    CommitHistoryConnectionTO history(CommitTO commitTO, String after, CommitAuthorTO author, String before, Integer first, Integer last, String path, String since, String until) throws Exception;

    @javax.validation.constraints.NotNull
    CommitConnectionTO parents(CommitTO commitTO, String after, String before, Integer first, Integer last) throws Exception;

}

File: src/main/kotlin-graphql/io/github/graphql/kotlin/resolver/CommitResolver.kt
package io.github.graphql.kotlin.resolver

import io.github.graphql.kotlin.model.*

/**
 * Resolver for Commit
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-07T19:48:14+0800"
)
interface CommitResolver {

    @Throws(Exception::class)
    fun associatedPullRequests(commitTO: CommitTO, after: String?, before: String?, first: Int?, last: Int?, orderBy: PullRequestOrderTO?): PullRequestConnectionTO?

    @Throws(Exception::class)
    fun blame(commitTO: CommitTO, path: String): BlameTO

    @Throws(Exception::class)
    fun comments(commitTO: CommitTO, after: String?, before: String?, first: Int?, last: Int?): CommitCommentConnectionTO

    @Throws(Exception::class)
    fun deployments(commitTO: CommitTO, after: String?, before: String?, environments: List<String>?, first: Int?, last: Int?, orderBy: DeploymentOrderTO?): DeploymentConnectionTO?

    @Throws(Exception::class)
    fun history(commitTO: CommitTO, after: String?, author: CommitAuthorTO?, before: String?, first: Int?, last: Int?, path: String?, since: String?, until: String?): CommitHistoryConnectionTO

    @Throws(Exception::class)
    fun parents(commitTO: CommitTO, after: String?, before: String?, first: Int?, last: Int?): CommitConnectionTO

}
